bin/lumen

```diff
@@ -1,40 +1,53 @@
-#!/bin/sh
+#!/bin/dash
+
+cwdir="$(pwd)"
 
-dir="$(pwd)"
 bin="$0"
-while [ -L "${bin}" ]
-do
-    x="$(readlink "${bin}")"
-    cd "$(dirname "${bin}")"
-    bin="${x}"
-done
-cd "$(dirname "${bin}")"
-home="$(pwd)"
-cd "${dir}"
-
-if [ ! -z "${LUMEN_HOST}" ]
-then
-    host="${LUMEN_HOST}"
-elif luajit -v > /dev/null 2>&1
-then
-    host=luajit
-elif lua -v > /dev/null 2>&1
-then
-    host=lua
-elif node -v > /dev/null 2>&1
-then
-    host=node
+
+die() {
+	echo "$*" 1>&2
+	exit 1
+}
+
+realpath() {
+	while [ -L "${bin}" ]; do
+		x="$(readlink "${bin}")"
+		cd "$(dirname "${bin}")" || die "Err: could not cd to '$bin' dir"
```

Comment:
For these simple operations like `cd', it's simpler and more idiomatic to let the command itself fail and present its own error message. This patch is also inconsistent in this, with some invocations of [ -L ... ] getting error treatment while others don't.